### DISCUSSION TITLE
Customize Badges by adding numbers on corner.

### DISCUSSION THREAD

**Discussion by Yawan-1** [mantinedev/mantine#5956] [Feature requests]
Hi,

I was working on implementing badges on my app and noticed that the feature to display numbers on badges is not available. I suggest adding the ability to display numbers on badges, specifically on the corner or right side, with additional customization options. Additionally, it would be beneficial to include a property on the badge number called "max" or **"maxCount"** to display **"100+"** or **"99+"** when the number exceeds a certain threshold.

Thank you for this wonderful Library.

<small>If this feature is already available and I missed it, I apologize. Please provide the link.</small>

**Comment by Kenzo-Wada** [comment]
In my opinion, using rightSection is sufficient for that needs since the usage is limited.

ex. this function will help you to show number like 99+


```js
function displayNumber(number) {
  if (number >= 100) {
    return '99+';
  } else {
    return number.toString();
  }
}
```

does this help what you want to do?

**Comment by Yawan-1** [comment]
The functionality you suggested will kick in when a number icon appears on top of another icon, similar to an unread notification dot on the top right corner. This function will be implemented within the library, not directly for the user's interaction. That's why I mentioned adding a `prop` of `maxCount`.

For now, I managed to add number/dot on top right corner of the icon/avatar by nesting Badge inside the Avatar component.

```
<Avatar  style={{ position: 'relative' }} src="image.jpg">
    <Badge style={{position: "absolute", top: "-1.5rem", right: "-1rem"}}>
        // Dot icon
    </Badge>
</Avatar>
```

To understand better let me explain more, I suggested two functionalities.
1. Number or dot badge to be placed on top right/left corner of any icon.
2. A `prop` called `maxCount` to the Badge component. It allows specifying the maximum number at which the Badge will start showing a "+" symbol, for example, displaying "99+" when the value inside the Badge exceeds that maxCount.

More like
```
<Avatar>
    <Badge content={57} maxCount={99}>
    </Badge>
</Avatar>
```

**Reply by rtivital** [reply]
https://mantine.dev/core/indicator/

**Comment by Yawan-1** [comment]
That's what I was talking about. Thanks.
